### PR TITLE
Enzyme Tests

### PR DESCRIPTION
**Pull request by atomiomi**

### CODE REVIEW

File: test/base/index.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  });
+
+  it('getValidation', () => {
+    const props = {
+      required: true,
+      pattern: '[0-9]*',
+      validate: (value) => value.length === 5,
+    };
+
+    const instance = shallow(React.createElement(Child, props)).instance();
```
Comment (suprMax) [inline test/base/index.js]: import `createElement` from React - less typing

File: test/base/index.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    assert.equal(instance.getError(), null, 'error was cleared');
+    assert.equal(callback.callCount, 1, 'callback was called');
+  });
+
+  it('actionValidate', (done) => {
+    const callback = fake();
+    const props = { required: true, error: 'Error' };
+    const instance = shallow(React.createElement(Child, props)).instance();
+    const action = instance.actionValidate(callback);
```
Comment (suprMax) [inline test/base/index.js]: could probably update actionValicate to return promise - easier to test then. ok as it is too though

File: test/base/index.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+
+const Base = require('../../lib/base').default;
+
+
+class Child extends Base {
+  render() { return null; }
+}
+
+describe('Base', () => {
```
Comment (suprMax) [inline test/base/index.js]: test if component connects to form?

File: test/base/index.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    const doneFake = fake();
+    const instance = shallow(React.createElement(Child, props)).instance();
+
+    instance.setError('Error', doneFake);
+    assert.equal(instance.getError(), 'Error', 'error was set');
+    assert.equal(props.onError.callCount, 1, 'on error was called');
+    assert.equal(doneFake.callCount, 1, 'callback was called');
+  });
+
```
Comment (suprMax) [inline test/base/index.js]: test `setPristine` too. there were some issues before with `select` components.

Comment (suprMax) [reply test/base/index.js]: also should probably test that `reset` works

File: test/input/index.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+const React = require('react');
+const { assert } = require('chai');
+const { mount } = require('enzyme');
+
+
+const Input = require('../../lib/input').default;
```
Comment (suprMax) [inline test/input/index.js]: test base input component too? there are some methods to test:
`getDefaultState`
`getValue`
`getCaretPosition`
`updateValue` 
check that `readDomValue` is called etc.

File: test/input/index.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+describe('<Input />', () => {
```
Comment (suprMax) [inline test/input/index.js]: test for custom callbacks too - `onEnterKey` and `onShiftEnterKey`

File: test/base/index.js (hunk 2 above)
Comment (atomiomi) [reply test/base/index.js]: let's change when there is a case

**Pull request merged**